nixos/modules/system/activation/switch-to-configuration.pl

```diff
@@ -124,10 +64,41 @@ sub boolIsTrue {
 }
 
 sub recordUnit {
-    my ($fn, $unit) = @_;
+    my ($action, $fn, $unit) = @_;
     write_file($fn, { append => 1 }, "$unit\n") if $action ne "dry-activate";
 }
 
+sub pathToUnitName {
+    my ($curSystemd, $path) = @_;
+    # Use current version of systemctl binary before daemon is reexeced.
+    open my $cmd, "-|", "$curSystemd/systemd-escape", "--suffix=mount", "-p", $path
+        or die "Unable to escape $path!\n";
+    my $escaped = join "", <$cmd>;
+    chomp $escaped;
+    close $cmd or die;
+    return $escaped;
+}
+
+sub unique {
```

Comment:
This function is unused and will be removed in the next iteration.

Reply:
Could use `uniq` from `List::MoreUtils` instead, or something like:
```
sub uniq {
  my %seen;
  return grep { !$seen{$_}++ } @_;
}
```
(From https://perlmaven.com/unique-values-in-an-array-in-perl)

Reply:
Yeah. but it is not used in the script. so there is not point in improving it, no?

Reply:
Yeah you're right - and out of scope of this PR.

Reply:
Fixed in https://github.com/Mic92/nixpkgs/commit/8ca649124dfeca49660aceeb0f37de7180dfc285
but let's introduce this later.